src/GridFS/StreamWrapper.php

```diff
@@ -338,6 +330,42 @@ public function url_stat(string $path, int $flags)
         return $this->stream_stat();
     }
 
+    /** @psalm-return ($mode == 'r' ? array{collectionWrapper: CollectionWrapper, file: object} : array{collectionWrapper: CollectionWrapper, filename: string, options: array}) */
```

Comment:
Not sure if this is relevant, but the ternary here only checks for "r" and `stream_open()` may call this with "rb". Can you invoke functions here, or are you restricted on syntax here?

Reply:
The logical `or` works. I also added the `@return` annotation so PHPStorm understands it.